### PR TITLE
Replaced Ginkgo and Gomega with Testify in functional tests

### PR DESCRIPTION
**Pull request by hbelmiro**
## The issue resolved by this Pull Request:
Resolves https://issues.redhat.com/browse/RHOAIENG-4261

## Description of your changes:

Replaced Ginkgo and Gomega with Testify in functional tests

## Testing instructions

Run `make test`. All tests must pass.

## Checklist
- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by dsp-developers** [comment]
A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-628`
An OCP cluster where you are logged in as cluster admin is required.

To use this image run the following:

```bash
cd $(mktemp -d)
git clone git@github.com:opendatahub-io/data-science-pipelines-operator.git
cd data-science-pipelines-operator/
git fetch origin pull/628/head
git checkout -b pullrequest 7ed99727f215de4fd1a0fb122b0e434f04b634e8
oc new-project opendatahub
make deploy IMG="quay.io/opendatahub/data-science-pipelines-operator:pr-628"
```

More instructions [here](https://github.com/opendatahub-io/data-science-pipelines-operator#deploy-dsp-instance) on how to deploy and test a Data Science Pipelines Application.

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-628`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-628`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-628`

**Comment by DharmitD** [comment]
@diegolovison can you also take a look, since you might need to update the related script that runs downstream. (I believe you should be good though since the Makefile command remains the same, but if you could just check and confirm)

File: controllers/suite_test.go [hunk 1]
```diff
@@ -20,36 +20,34 @@ package controllers
 
 import (
 	"context"
-	"path/filepath"
-	"testing"
-	"time"
-
+	"github.com/go-logr/logr"
+	"github.com/onsi/gomega/format"
```
Comment (DharmitD) [inline controllers/suite_test.go:24]: Do we still need this import?

File: go.mod [hunk 2]
```diff
@@ -12,7 +12,6 @@ require (
 	github.com/manifestival/controller-runtime-client v0.4.0
 	github.com/manifestival/manifestival v0.7.2
 	github.com/minio/minio-go/v7 v7.0.56
-	github.com/onsi/ginkgo/v2 v2.9.5
 	github.com/onsi/gomega v1.27.7
```
Comment (DharmitD) [inline go.mod:15]: gomega could be taken out from here as well, just make sure to run a `go mod tidy` locally.

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-628`

File: controllers/suite_test.go (hunk 1 above)
Comment (hbelmiro) [reply controllers/suite_test.go]: We don't. Removed.

File: go.mod (hunk 2 above)
Comment (hbelmiro) [reply go.mod]: Done.

**Review by DharmitD — COMMENTED** [review]
/lgtm

**Comment by VaniHaripriya** [comment]
Verified the tests in local. /lgtm

**Pull request merged**